sql/hive/src/main/java/org/apache/hadoop/hive/ql/io/orc/VectorizedSparkOrcNewRecordReader.java

```diff
@@ -0,0 +1,318 @@
+
+    this.columnIDs = new ArrayList<>(columnIDs);
+    this.internalValue = this.reader.createValue();
+    this.progress = reader.getProgress();
+    this.row = new Row(this.internalValue.cols, this.columnIDs);
+  }
+
+  @Override
+  public void close() throws IOException {
+    reader.close();
+  }
+
+  @Override
+  public NullWritable getCurrentKey() throws IOException,
+      InterruptedException {
+    return NullWritable.get();
+  }
+
+  @Override
+  public InternalRow getCurrentValue() throws IOException,
+      InterruptedException {
+    if (indexOfRow >= numRowsOfBatch) {
+      return null;
+    }
+    row.rowId = indexOfRow;
+    indexOfRow++;
+
+    return row;
+  }
+
+  @Override
+  public float getProgress() throws IOException, InterruptedException {
+    return progress;
+  }
+
+  @Override
+  public void initialize(InputSplit split, TaskAttemptContext context)
+      throws IOException, InterruptedException {
+  }
+
+  @Override
+  public boolean nextKeyValue() throws IOException, InterruptedException {
+    if (indexOfRow == numRowsOfBatch) {
+      if (reader.next(NullWritable.get(), internalValue)) {
+        if (internalValue.endOfFile) {
+          progress = 1.0f;
+          numRowsOfBatch = 0;
+          indexOfRow = 0;
+          return false;
+        } else {
+          assert internalValue.numCols == numColumns : "Incorrect number of columns in OrcBatch";
+          numRowsOfBatch = internalValue.count();
+          indexOfRow = 0;
+          progress = reader.getProgress();
+        }
+        return true;
+      } else {
+        return false;
+      }
+    } else {
+      if (indexOfRow < numRowsOfBatch) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+  }
+
+  /**
+   * Adapter class to return an internal row.
+   */
+  public static final class Row extends InternalRow {
+    protected int rowId;
+    private List<Integer> columnIDs;
+    private final ColumnVector[] columns;
+
+    private Row(ColumnVector[] columns, List<Integer> columnIDs) {
+      this.columns = columns;
+      this.columnIDs = columnIDs;
+    }
+
+    @Override
+    public int numFields() { return columnIDs.size(); }
+
+    @Override
+    public boolean anyNull() {
+      for (int i = 0; i < columns.length; i++) {
+        if (columnIDs.contains(i)) {
+          if (columns[i].isRepeating && columns[i].isNull[0]) {
+            return true;
+          } else if (!columns[i].isRepeating && columns[i].isNull[rowId]) {
+            return true;
+          }
+        }
+      }
+      return false;
+    }
+
+    @Override
+    public boolean isNullAt(int ordinal) {
+      ColumnVector col = columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return col.isNull[0];
+      } else {
+        return col.isNull[rowId];
+      }
+    }
+
+    @Override
+    public boolean getBoolean(int ordinal) {
+      LongColumnVector col = (LongColumnVector)columns[columnIDs.get(ordinal)];
```

Comment:
nit: space after `(LongColumnVector)`. This might also apply to other places in the PR but I am not pointing out each instance.